### ISSUE TITLE
OSX version does not start

### ISSUE THREAD

**Issue by pramsky** [borgbase/vorta#79]
Hello!

I am unable to start Vorta (0.5.2). Clicking the app or using open -a Vorta.app does nothing. Running the binary under Contents/MacOS/vorta returns the following.

`[43076] Error loading Python lib '/var/folders/zh/0t27d1hj01gcs6j5sfdb5tqh0000gn/T/_MEIiht4jn/libpython3.6m.dylib': dlopen: dlopen(/var/folders/zh/0t27d1hj01gcs6j5sfdb5tqh0000gn/T/_MEIiht4jn/libpython3.6m.dylib, 10): Symbol not found: _futimens
  Referenced from: /var/folders/zh/0t27d1hj01gcs6j5sfdb5tqh0000gn/T/_MEIiht4jn/libpython3.6m.dylib
  Expected in: /usr/lib/libSystem.B.dylib
 in /var/folders/zh/0t27d1hj01gcs6j5sfdb5tqh0000gn/T/_MEIiht4jn/libpython3.6m.dylib`

OS: Sierra 10.12.6
Python version: system python 2.7 , 3.7 also available on system

**Comment by m3nu** [comment]
Tricky. I'm building the app on macOS 10.14. Looks like it's linking to some system libs that aren't available on 10.12.

Will try in a virtual machine.

**Comment by m3nu** [comment]
Since you have a new Python and know the CLI, you can also use the pip version:

`pip install vorta`  (or `pip install --user vorta`)

Then just make sure Borg is also available. (`pip install borgbackup` or `brew install borgbackup`)

**Comment by m3nu** [comment]
This says that to support older versions, it needs to be built on them. https://stackoverflow.com/questions/49908236/pyinstaller-executable-fails-on-old-os-x

I think we can support 2 macOS versions. For anything older, it can be installed via `pip`. I tested 10.13 some time ago, but don't have 10.12 around any more.

**Comment by ThomasWaldmann** [comment]
See borg's Vagrantfile.

**Comment by m3nu** [comment]
To package for different macOS versions? I already copied your Vagrantfile in the beginning. :-)

I'd rather have a fix that lets the same file run on all versions. And again, 10.12 is 2 versions back now. Not many people will run it.

**Comment by ThomasWaldmann** [comment]
There's a rather old osx box. Building on that might result in a binary that works on all versions.

**Comment by m3nu** [comment]
True. Worth a try.

**Comment by pramsky** [comment]
Attempted to run vorta directly after cloning the repo and running `pip3 install -e .` and `pip3 install -r requirements-dev.txt` and `/usr/local/bin/vorta`

Result:
```
2018-12-05 09:42:04.707 Python[81377:28803932] Metal pipeline state creation failed with error: Error Domain=MTLLibraryErrorDomain Code=3 "Compiler encountered an internal error" UserInfo={NSLocalizedDescription=Compiler encountered an internal error}
2018-12-05 09:42:04.721 Python[81377:28803932] Metal pipeline state creation failed with error: Error Domain=MTLLibraryErrorDomain Code=3 "Compiler encountered an internal error" UserInfo={NSLocalizedDescription=Compiler encountered an internal error}```

**Comment by m3nu** [comment]
I suspect the binary package for [PyQt](https://pypi.org/project/PyQt5/) (which does all the graphics stuff) is not compatible with your macOS version either. ("Metal" is a new graphics framework that Apple has been using to replace OpenGL.) This other project [recommends](https://www.pmel.noaa.gov/maillists/tmap/ferret_users/fu_2018/msg00282.html) using the binaries provided by the Anaconda Python distribution. They seem to run on older Macs as well.

**Comment by m3nu** [comment]
I'll package a Vorta version with Anaconda and see how it works. They do their own compiling which is often more backwards-compatible than normal wheels.

**Comment by m3nu** [comment]
Not much luck with Qt on Conda. The tabs look better, but I keep get the same Metal error that you get (no crash though). It's also not working after freezing it. Their Qt version is very outdated as well.

**Comment by pramsky** [comment]
hmm. I am upgrading to High Sierra anyways. I'll see if it has any issues there.

**Comment by m3nu** [comment]
Yeah. Let me know how it goes.

I found that my GF has 10.12 as well. Same error. I may try to build the official release on 10.12 and see if it runs everywhere.

**Comment by m3nu** [comment]
> When building binaries on some random machine that is also used for other stuff, there's always some risk involved.

Agree. This was just for testing. Now that you improved the pyinstaller settings, we can look into building for multiple platforms.

**Comment by pramsky** [comment]
@m3nu 

Sorry for the late update, works perfectly on High Sierra. I have been running it every 4 hours for the past few days now.

**Comment by m3nu** [comment]
Thanks, but it won't work. Already tested. We'll probably end up building the fat binary for macOS on *oldstable* in order to support *stable* and *oldoldstable*.
